client/transaction-pool/src/api.rs

```diff
@@ -253,4 +286,16 @@ impl<Client, F, Block> sc_transaction_graph::ChainApi for LightChainApi<Client,
 			Ok(Some(transactions))
 		}.boxed()
 	}
+
+	fn leaves(&self) -> Result<Vec<Block::Hash>, Self::Error> {
+		self.backend.blockchain().leaves().map_err(Into::into)
+	}
+
+	fn tree_route(
```

Comment:
Why not merge the two methods into `find_retracted_blocks`? This logic to do that does not sound like transaction pool at all.